src/test/java/io/jenkins/plugins/analysis/warnings/IssuesRecorderRunAlwaysITest.java

```diff
@@ -0,0 +1,161 @@
+package io.jenkins.plugins.analysis.warnings;
+
+import java.io.IOException;
+import java.util.Collections;
+
+import org.assertj.core.api.Assertions;
+import org.junit.Test;
+
+import com.google.errorprone.annotations.CanIgnoreReturnValue;
+
+import static hudson.remoting.Launcher.isWindows;
+import io.jenkins.plugins.analysis.core.model.AnalysisResult;
+import static io.jenkins.plugins.analysis.core.model.Assertions.assertThat;
+import io.jenkins.plugins.analysis.core.model.StaticAnalysisTool;
+import io.jenkins.plugins.analysis.core.model.StaticAnalysisToolSuite;
+import io.jenkins.plugins.analysis.core.steps.IssuesRecorder;
+import io.jenkins.plugins.analysis.core.steps.ToolConfiguration;
+import io.jenkins.plugins.analysis.core.testutil.IntegrationTest;
+import io.jenkins.plugins.analysis.core.views.ResultAction;
+
+import hudson.model.Action;
+import hudson.model.FreeStyleBuild;
+import hudson.model.FreeStyleProject;
+import hudson.model.Result;
+import hudson.tasks.Shell;
+
+/**
+ * Integration tests of the run always option for a freestyle jobs.
+ *
+ * @author Martin Weibel
+ */
+public class IssuesRecorderRunAlwaysITest extends IntegrationTest {
+
+    /**
+     * Runs TestBuild with FAILURE, should still run checkstyle because of runAways is activated.
+     */
+    @Test
+    public void shouldRunEvenResultIsFailure() {
+        FreeStyleProject project = createJob();
+
+        project.getBuildersList().add(new Shell("exit 1"));
+
+        enableCheckStyle(project, true);
+
+        AnalysisResult result = scheduleBuildAndAssertStatus(project, Result.FAILURE).getResult();
+
+        assertThat(result).hasTotalSize(0);
+        assertThat(result).hasErrorMessages(
+                "No files found for pattern '**/checkstyle-result.xml'. Configuration error?");
+    }
+
+    /**
+     * Runs TestBuild with FAILURE, should not run checkstyle because runAways is disabled.
+     */
+    @Test
+    public void shouldNotRunWhenResultIsFailure() {
+        FreeStyleProject project = createJob();
+
+        project.getBuildersList().add(new Shell("exit 1"));
+        enableCheckStyle(project, false);
+
+        ResultAction result = scheduleBuildAndAssertStatus(project, Result.FAILURE);
+
+        Assertions.assertThat(result).isNull();
```

Comment:
Methoden sollten niemals null zurückliefen. Bekommen Sie da keine Warnung in IntelliJ? Da braucht es eine passende Validierung.